tools/common/Assembly.cs

```diff
@@ -336,6 +336,27 @@ void AddFramework (string file)
 				Driver.Log (3, "Linking with the framework {0} because it's referenced by a module reference in {1}", file, FileName);
 		}
 
+		public string GetCompressionLinkingFlag ()
+		{
+			switch(App.Platform) {
+			case ApplePlatform.MacOSX:
+				if (App.SdkVersion >= new Version (10, 11, 0))
+					return "-lcompression";
+				return "-weak-lcompression";
+			case ApplePlatform.TVOS:
+			case ApplePlatform.iOS:
+				if (App.SdkVersion >= new Version (9,0))
+					return "-lcompression";
+				return "-weak-lcompression";
+			case ApplePlatform.WatchOS:
+				if (App.SdkVersion >= new Version (2, 0))
+					return "-lcompression";
+				return "-weak-lcompression";
```

Comment:
2.0 is the minimum for watchOS so it's always supported

Reply:
👍